### PR TITLE
Feat/badgesbackend1

### PR DESCRIPTION
**Pull request by pieryvdl**
Hi guys,

So me and Lerato worked on the backend for the badges, they work, we have tested them. 
You'll notice a new badge service for all of the badges, that works with Firebase. We will explain how it works on Monday. 
Then you'll also notice I made some of the service singletons, this is to keep clean code throughout our app and not duplicate instances of these services, which will hopefully improve performance.

### CODE REVIEW

**Comment by pieryvdl** [comment]
@u22520016 Please send me your unique uid so that I can create a badge profile in Firebase for you to make testing easier. @MonikaTheiss I am making one for you since I know your account uid.

File: lib/services/badges_S/badge_service.dart [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+            }
+
+            if (count == 10) {
+              await _firestore
+                  .collection("badges")
+                  .doc(currentUser.uid)
+                  .update({
+                //if the count is == 10 then update the document with the unlocked status
+                "collector_badge": {
+                  //CHANGE
```
Comment (MonikaTheiss) [inline lib/services/badges_S/badge_service.dart:121]: Not too sure what this comment means, just wanted to point it out incase you left this comment as a reminder to chane something 😊

**Review by MonikaTheiss — COMMENTED** [review]
Code looks quite exciting 😄 Left a small comment, not too sure if thats something you still need to do. Will test on my side now 😊 Thank you @LeratoKgomo and @pieryvdl for working so hard on this !

**Comment by u22520016** [comment]
> @u22520016 Please send me your unique uid so that I can create a badge profile in Firebase for you to make testing easier. @MonikaTheiss I am making one for you since I know your account uid.

I don't really have my own account I test with, I just use the account Im always logged in with.  Can I use the Monika account one that Monika sent on discord ages ago?

**Review by MonikaTheiss — APPROVED** [review]
Awesome ! Thank you so much @LeratoKgomo and @pieryvdl ! I'm very happy with the changes you made. Well done on figuring out the backend for the badges, I love how easy it is to understand 😄

**Comment by pieryvdl** [comment]
> I don't really have my own account I test with, I just use the account Im always logged in with. Can I use the Monika account one that Monika sent on discord ages ago?

Hi Franco, yes you can use Monika's account, I will reset it for you on Firebase.

**Review by MonikaTheiss — APPROVED** [review]

**Review by u22520016 — COMMENTED** [review]
The code looks good, I'm just a little unsure why the unlockNightOwlBadge function is on multiple pages, but not all.  Should it not just be on the home page since that will be the first page that pops up when the user opens the app?

**Comment by pieryvdl** [comment]
The reason for it not being on all pages is that we don't want to overwhelm Firebase with calls after 10 pm, the reason why it is not just on the homepage is because someone cna open the app before 10 pm, use it and they never navigate to the homepage, i.e. we need to have the nightOwlBadge on more pages, but not all, because of the reason above.

**Comment by u22520016** [comment]
> The reason for it not being on all pages is that we don't want to overwhelm Firebase with calls after 10 pm, the reason why it is not just on the homepage is because someone cna open the app before 10 pm, use it and they never navigate to the homepage, i.e. we need to have the nightOwlBadge on more pages, but not all, because of the reason above.

Ohh okay I understand.  In my head it was when you open the app after 10, but your reasoning makes more sense.

**Pull request merged**